doc/theoretical_description_metrics.rst

```diff
@@ -0,0 +1,264 @@
+
+.. math::
+
+   \text{HSIC} = \operatorname{trace}(\mathbf{H} \mathbf{K} \mathbf{H} \mathbf{L})
+
+where:
+
+- :math:`\mathbf{K}` and :math:`\mathbf{L}` are the kernel matrices representing the interval sizes and coverage indicators, respectively.
+- :math:`\mathbf{H}` is the centering matrix, :math:`\mathbf{H} = \mathbf{I} - \frac{1}{n} \mathbf{11}^\top`.
+
+This measure is crucial for determining whether certain sizes of prediction intervals are systematically more or less likely to contain the true values, which can highlight biases in interval-based predictions.
+
+
+Coverage Width-Based Criterion (CWC)
+------------------------------------
+
+The **Coverage Width-Based Criterion (CWC)** evaluates prediction intervals by balancing their empirical coverage and width. It is designed to both reward narrow intervals and penalize those that do not achieve a specified coverage probability [6].
+
+.. math::
+
+   \text{CWC} = (1 - \text{Mean Width Score}) \times \exp\left(-\eta \times (\text{Coverage Score} - (1-\alpha))^2\right)
+
+
+
+Regression MWI Score
+--------------------
+
+The **Regression MWI (Mean Winkler Interval) Score** evaluates prediction intervals by combining their width with a penalty for intervals that do not contain the observation [8, 10].
+
+.. math::
+
+   \text{MWI Score} = \frac{1}{n} \sum_{i=1}^{n} (y_{\text{pred, up}}^{(i)} - y_{\text{pred, low}}^{(i)}) + \frac{2}{\alpha} \sum_{i=1}^{n} \max(0, |y_{\text{true}}^{(i)} - y_{\text{pred, boundary}}^{(i)}|)
+
+where :math:`y_{\text{pred, boundary}}^{(i)}` is the nearest interval boundary not containing :math:`y_{\text{true}}^{(i)}`, and :math:`\alpha` is the significance level.
+
+
+
+2. Calibration metrics
+======================
+
+Expected Calibration Error (ECE)
+--------------------------------
+
+**Expected Calibration Error (ECE)** measures the difference between predicted probabilities of a model and the actual outcomes, across different bins of predicted probabilities [7].
+
+.. math::
+
+   \text{ECE} = \sum_{b=1}^{B} \frac{n_b}{n} | \text{acc}(b) - \text{conf}(b) |
+
+where:
+
+- :math:`B` is the total number of bins,
+- :math:`n_b` is the number of samples in bin :math:`b`,
+- :math:`\text{acc}(b)` is the accuracy within bin :math:`b`,
+- :math:`\text{conf}(b)` is the mean predicted probability in bin :math:`b`.
+
+
+Top-Label Expected Calibration Error (Top-Label ECE)
+----------------------------------------------------
+
+**Top-Label ECE** focuses on the class predicted with the highest confidence for each sample, assessing whether these top-predicted confidences align well with actual outcomes. It is calculated by dividing the confidence score range into bins and comparing the mean confidence against empirical accuracy within these bins [5].
+
+.. math::
+
+   \text{Top-Label ECE} = \sum_{b=1}^{B} \frac{n_b}{n} \left| \text{acc}_b - \text{conf}_b \right|
```

Comment:
The difference with the previous metric is not clear (the equations are too similar).

Reply:
Indeed, I think I copy-pasted the same one. Let me update this.

Reply:
done